File: 4107-find-missing-elements/4107-find-missing-elements.py
class Solution(object):
    def findMissingElements(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        mini = min(nums)
        maxi = max(nums)

        n = len(nums)
        s = set(nums)
        res = []

        for i in range(mini , maxi+1):
            if i not in s:
                res.append(i)

        return res

File: 4107-find-missing-elements/4107-find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        vector<int> res;
        int mini = INT_MAX;
        int maxi = INT_MIN;

        unordered_set<int> ans;
        for(auto num : nums){
            mini = min(num ,mini);
            maxi = max(num,maxi);
            ans.insert(num);
        }

        for(int i = mini; i < maxi+1; i++){
            if(ans.find(i) == ans.end()) res.push_back(i);
        }
        return res;
    }
};